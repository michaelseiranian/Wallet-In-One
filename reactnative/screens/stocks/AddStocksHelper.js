import { auth_post } from "../../authentication";

/**
 * Custom react hook that is used to connect to the backend when the user has succesfully added an account via the SDK
 */
export default function AddStocksHelper(){

    /**
    * Function that takes in a temporary public token which is then exchanged for an access token for connecting to endpoints
    * @param publicToken
    * @returns access_token
    */
    const getAccessToken = async (publicToken) => {
        const body = {
          public_token: publicToken
        }
        const response = await auth_post('/stocks/get_access_token/', body)
        return response.body.access_token
    }

    /**
     * Function that accesses the balance of as stock account by using the access token generated
     * @param accessToken 
     * @returns balance
     */
    const getBalance = async (accessToken) => {
        const body = {
            access_token: accessToken
        }
        const response = await auth_post('/stocks/get_balance/', body)
        const data = response.body;
        return (parseFloat(data.accounts[0].balances.current)*0.83).toFixed(2)
    }

    /**
     * Function that accesses the transactions for a stock account by using the access token generated
     * @param accessToken 
     * @returns transactions
     */
    const getTransaction = async (accessToken) => {
        const body = {
          access_token: accessToken
        }
        const response = await auth_post('/stocks/get_transactions/', body)
        return response.body
    }

    /**
     * Function that adds a stock account to the backend database
     * @param {*} account 
     * @param {*} success 
     * @param {*} access_token 
     * @param {*} balance 
     * @param {*} image 
     */
    const addAccount = async (account, success, access_token, balance, image) => {
        const account_data = {
          account_id: account._id,
          name: account.meta.name,
          institution_name: success.metadata.institution.name,
          institution_id: success.metadata.institution.id,
          access_token: access_token,
          balance: balance,
          institution_logo: image,
        }
        const response = await auth_post('/stocks/add_stock_account/', account_data);
        return response.status;
    };

    /**
     * Function that adds a transaction to the backend database
     * @param {*} element 
     * @param {*} fetched_transaction_list 
     */
    const addTransaction = async (element, fetched_transaction_list) => {
        let latitude = parseFloat(((Math.random() * (7) + 35.5).toFixed(3))) // Randomly Generate Location since Plaid does not provide transaction location specifically for Investment Transactions
        let longitude = parseFloat(((Math.random() * (43) + 77).toFixed(3))) * -1
        const body = {
          account_id: element.account_id,
          investment_transaction_id: element.investment_transaction_id,
          security_id: element.security_id,
          date: element.date,
          name: element.name,
          quantity: element.quantity,
          amount: element.amount * 0.83, // to convert to GBP
          price: element.price,
          fees: element.fees,
          stock: fetched_transaction_list.accounts[0].account_id,
          latitude: latitude,
          longitude: longitude
        }
        await auth_post('/stocks/add_transaction_account/', body)
    };

    /**
     * Function that adds a Stock to the backend database
     * @param {*} stock 
     * @param {*} stockInfo 
     */
    const addStock = async (stock, stockInfo) => {
        const body = {
          institution_price: (stock.institution_price).toFixed(2),
          quantity: stock.quantity,
          name: stockInfo.name,
          ticker_symbol: stockInfo.ticker_symbol,
          stockAccount: stock.account_id,
          security_id: stockInfo.security_id
        }
        await auth_post('/stocks/add_stock/', body)
    }

    /**
     * Function that accesses the logo of the specified institution
     * @param {*} success 
     * @returns logo
     */
    const getLogo = async (success) => {
        const body = {
          name: success.metadata.institution.name
        }
        const response = await auth_post('/stocks/get_logo/', body)
        return response.body.logo
    }

    /**
     * Function that accesses all stock holdings a specified stock account
     * @param {*} accessToken 
     * @returns stocks
     */
    const getStocks = async (accessToken) => {
        const body = {
          access_token: accessToken
        }
        const response = await auth_post('/stocks/get_stocks/', body)
        const data = response.body;
        return [data.holdings, data.securities]
    }

    return { getAccessToken, getBalance, getTransaction, addAccount, addTransaction, addStock, getLogo, getStocks };
}