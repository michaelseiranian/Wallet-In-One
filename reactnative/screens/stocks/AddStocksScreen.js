import React, { useState, useEffect,useRef  } from 'react';
import { Image } from 'react-native';
import PlaidLink from '@burstware/expo-plaid-link'
import { useIsFocused } from '@react-navigation/native';
import { auth_post } from '../../authentication';
import { useTheme } from "reactnative/src/theme/ThemeProvider";
import AddStocksHelper from './AddStocksHelper';
import { styles } from "reactnative/screens/All_Styles.style.js";
import ConditionalModal from '../Modal';

import { StyleSheet, Pressable, View, Animated} from 'react-native';

/**
 * Component that displays the Plaid SDK for connecting Stock Accounts to the app
 */

const PlaidComponent = ({ navigation }) => {
  const [linkToken, setLinkToken] = useState('');
  const isFocused = useIsFocused();
  const {getAccessToken, getBalance, getTransaction, addAccount, addTransaction, addStock, getLogo, getStocks} = AddStocksHelper();
  let access_token = ''
  let balance = ''
  let image = ''
  let fetched_transaction_list = null
  let stocks = null
  let securities = null

  let data_response = null

  const stylesInternal = StyleSheet.create({
    modalView: {
      margin: 20,
      backgroundColor: 'white',
      borderRadius: 20,
      padding: 35,
      alignItems: 'center',
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 4,
      elevation: 5,
    },
    button: {
      borderRadius: 20,
      padding: 10,
      elevation: 2,
    },
    buttonClose: {
      backgroundColor: '#2196F3',
    },
    textStyle: {
      color: 'white',
      fontWeight: 'bold',
      textAlign: 'center',
    },
    modalText: {
      marginBottom: 15,
      textAlign: 'center',
    },
  });

  /**
   * Whenever the screen is in focus, a call to the backend is made to access a temporary link token to open the SDK.
   */
  useEffect(() => {
  const initiatePlaidLink = async () => {
      const response = await auth_post('/stocks/initiate_plaid_link/')
      if(response.status == 200){
        setLinkToken(response.body.link_token)
      }
  };
  if(useIsFocused){initiatePlaidLink()}
}, [isFocused])


  const [modalVisible, setModalVisible] = useState(false);
  const [modalText, setModalText] = useState("Empty Modal");
  const scaleValue = useRef(new Animated.ValueXY({x: 0.5, y: 0.5})).current;

  const {dark, colors, setScheme } = useTheme();

  return (
    <>

      {
        linkToken !== undefined
          ?
          <PlaidLink
            linkToken={linkToken}
            onSuccess={async (success) => {
              let account_list = success.metadata.accounts
              access_token = await getAccessToken(success.publicToken)
              balance = await getBalance(access_token)
              let stock_response = await getStocks(access_token)
              stocks = stock_response[0]
              securities = stock_response[1]

              account_list.forEach(async element => {
                image = await getLogo(success)
                data_response = await addAccount(element, success, access_token, balance, image)

                if(data_response != 400){
                  fetched_transaction_list = await getTransaction(access_token)
                  fetched_transaction_list.investment_transactions.forEach(element => {addTransaction(element, fetched_transaction_list)})

                  stocks.forEach(element => {
                    let stockInfo = securities[stocks.indexOf(element)]
                    addStock(element, stockInfo)
                  })
                  setModalText("Stock account has been successfully added.")
                  setModalVisible(true);
                }else{
                  setModalText("Stock account has already been added!")
                  setModalVisible(true);
                }
              });
            }}
          />
          :
          <View />
      }



  <View>


  <ConditionalModal
    headerText={modalText}
    bodyText={
      <View style={stylesInternal.modalView}>
        {modalText == "Stock account has been successfully added." &&
          <Image
            style={{ width: 100, height: 100 }}
            source={{ uri: `https://cdn-icons-png.flaticon.com/512/4436/4436481.png` }}
          />
        }
        {modalText == "Stock account has already been added!" &&
          <View>
            <Image
              style={{ width: 100, height: 100 }}
              source={{ uri: `http://www.setra.com/hubfs/Sajni/crc_error.jpg` }}
            />
          </View>
        }
      </View>
    }

    visible={modalVisible}
    onClose={() =>navigation.navigate("Stock Account List")}
    cancelButtonName={"View Stock Accounts"}
    oneButton={true}
  />
  </View>
    </>
  );
};

export default PlaidComponent;