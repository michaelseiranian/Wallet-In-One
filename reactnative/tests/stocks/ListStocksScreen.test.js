import React from "react";
import { render, screen, fireEvent, act, waitFor} from '@testing-library/react-native';
import SuccessComponent from "../../screens/stocks/ListStocksScreen";
import stocks from './fixtures/stocks.json'
import transactions from './fixtures/transaction.json'

var navigationData = {
    "accessToken": "access-sandbox-9cf73c44-d4f5-4a9c-84ce-6203d35aa40d",
    "accountID": "ALxMKPjkkdtZVpLBdXx8ia9prXRBmwh67wXWd",
    "account_name": "Plaid IRA",
    "balance": "266.23",
    "balance_currency": "GBP",
    "logo": "iVBORw0KGgoAAAANSUhEUgAAAJgAAACYCAMAAAAvHNATAAAAOVBMVEVHcEyZHCCYGx////+ZGx+ZGx+ZHCCcICKZGx+cKCyTFRjdtLWpSEv58PDnysu4aGrx3t7SnZ7FhIaoTrn+AAAACnRSTlMAR///tZt1INj4D9SNwAAABVVJREFUeNrNnNl2ozAMhsGDsS22wPs/7JCkbYIDshZD7Ks5Z6D90aet3qpKO1zd2qZpvA/GBO/Xf9q2dtU3h1sVeXMw/KrPfUfUoaY3dReLqymiXuLqq2zFUPWj7Xy7CVRdoc1ZIx/BuhJlPcYp0vSyTpGWR1Z+aW0w2UZo86Utb7IOXxdGMS/POpgTRlAbrTUnDZ2nucacNhpXGkY9ztacPIQ4rTl9WImuxlwwmkJ18ZU5by4a3pWpi6nsQl2rsuL8i+1nF+siK7Pm8mGLyPfCGlCbr4xk3XThO8KCK83xiQHQmq+NtkAHS7rZtxws6WZRBoPESP4q5ouWCnLp0DEllME87o8uMGFGpRum8R82xpTB+nnYfbFjlvOPiASzDIiwoU+ZDPrb3osLsCJzz/NhwpQtSS+DsGP0IQDL/+3uNy8alusP2LFZx6vm7oBGp2B5/wGfX4YGjaPWIugHDcuVzkAnuVeZ3HHQ61ia2OQd/jWO2B0CZjISy5lD8sPLHBLznYpl7GS3kHjHEdtpgEnHcmaRjEzmUC+5aVjGBp+SbzhqG4a5/5xmeWPE5Ef693iSVLCMY6dLf0lDI3lPRUgxT9bLhUvynWXLzt50liMvJrcsPQ8Hg6WA5Fv3k+70xSwlJF8No+UmSQbLjk/ylco8N7S2vys3yT+WjpC+OxnLuGrQSP7GJeGPXKwsoSxFJH/jkjLtFG4iltFrRJK/Tkaa1+wk9VJK8ulkjvKkjGVUZckkn05GmkcBEcvopZms6+FktJlNpMUYDgiBmOTTyWgzYliLcWQJOclHh0Gc00dajMPfOEpJPryf+CjWLu4zitM+g6QxxKBMlKV9W0QVlkPyHpbUSUQwbJZykvcGgzzrirUYe5RUJNd8QV4H4bJUkVzzBXn+HJCytPNb48d5JNd8QZ/Yx8rSJycdSZYwrMX4tEf0GUySqzDOmimHJahIrhmWIYzFMjLvxBbGWHMABsvoI0YmSRMqztNIWYpZRo9ySa41iSWsp7KMjdvzhbGWj5AWI+rne/rfeAcoWTsZkLK0Zakm6ZnCkLK0ZTkqSXruki6NZfQBfJK8zE9nGT3GJ8kXRmQ5cmfdP4Xxtn9gLcaLZUzS8IflLs8jZenFUk9ybRS56/NIWfpjuS3gEpJra+24r6RZRmlfQvI+R2CysRzC7iMSkoa/iQ3SLDOQ9IJNM0iL8cNyK32UkGwE28WQFuNpnIjkAqJsIdjOg7QYD5Zbk4pIPqah2GGJlKXnOrye5HN22LNZDhjLCLWIpBfuSURZbg0qI2lFe8YgwXLUk2yrSuRkKMvtf4pI/i4MsvcJIwslSw6SXrrxFSlL3baWykha8QbA47I0TBlI1uI938csh1s+koLkj81i5IpJWVwiLUa2mJSxnCm6tCQlLPsrSApYouv3OpLb7YD8VLacRdJWlcpkBPdXdDwak83nkLTUrYCSyQINSac+KZL0MhFJW1VakyW9bMliMMHZh0RdEpG0OTbzg0H9fwRtDpOnfxTmBLqkrznxpt0ji1VJ3YkRZPd6Z/jC6oyHyw6VCWLS5jzMcmQzAUnszJTg+M+BMkF2rTMfmIJ+zEKyzX7EbO9oyY1NsjnjUB70800Zk8lDeaJTZgAQpm7QZNf6tIOfYN7Mxo7J9syjsm9m67JlsDxnLFek/TLyY/KKY88AZuqYMUk/9q87WA+GF9mMA+zlHvkv9pKEcq+VKPgijnKvLin3spdyr8cp90Khcq9gKvjSqnKv+Sr3YrSCr5Ir+PK9bNKaMm9SLPTuSXvqbZ1lXiJa8LWrJV9UW/LVvi9x5V2G/CbvnOuj/wNfYsKPslREjwAAAABJRU5ErkJggg==",
    "name": "Vanguard",
    "transactions": undefined
}

/**
 * Tests for the ListStocksScreen
 */
describe('<SuccessComponent />', () => {

    beforeEach(() => {
        global.fetch =  jest.fn( async (url) => {
            if(url.includes('list_accounts')){
            return Promise.resolve({
                status: 200, json: () => stocks.all
            })
            }
            else{
                return Promise.resolve({
                    status: 200, json: () => transactions.all
                })
            }
        })
      })

      afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })

      it('listStocksScreen snapshot test', async () => {
        const listStocksScreen = render(<SuccessComponent />);

            await waitFor( () => {
                const loading = screen.UNSAFE_queryByType('ActivityIndicator');
                expect(loading).toBeNull();
            })
        expect(listStocksScreen).toMatchSnapshot();
    }
    )

    it('listStocksScreen test touchable opacity', async () => {
        const navigate = jest.fn();
        const listStocksScreen = render(<SuccessComponent navigation={{ navigate }}/>);

            await waitFor( () => {
                const loading = screen.UNSAFE_queryByType('ActivityIndicator');
                expect(loading).toBeNull();
            })
        fireEvent.press(await screen.getByText('Plaid IRA'));
        expect(navigate).toHaveBeenCalledWith("StockAsset", navigationData);
    }
    )
})

describe('<SuccessComponent />', () => {
    beforeEach(() => {
        global.fetch =  jest.fn( async (url) => {
            return Promise.resolve({
                status: 200, json: () => null
            })
        })
      })

      afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })

      it('listStocksScreen empty data', async () => {
        const listStocksScreen = render(<SuccessComponent />);

        await waitFor( () => {
            const loading = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(loading).toBeNull();
        })
        expect(listStocksScreen).toMatchSnapshot();
      })
})