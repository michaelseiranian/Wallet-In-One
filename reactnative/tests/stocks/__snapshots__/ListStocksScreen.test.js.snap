// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessComponent /> listStocksScreen empty data 1`] = `
<View
  style={
    {
      "backArrow": {
        "color": "purple",
        "fontSize": 30,
        "fontWeight": "900",
        "paddingVertical": 10,
      },
      "btn": {
        "alignItems": "center",
        "borderRadius": 10,
        "margin": 10,
        "padding": 10,
        "width": "40%",
      },
      "button": {
        "backgroundColor": "purple",
        "borderRadius": 25,
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": "4%",
        "paddingHorizontal": "12%",
        "paddingVertical": "2%",
        "textAlign": "center",
        "width": "75%",
      },
      "buttonWide": {
        "backgroundColor": "purple",
        "borderRadius": 10,
        "marginVertical": 10,
        "padding": 20,
      },
      "container": {
        "backgroundColor": "white",
        "flex": 1,
      },
      "input": {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "height": 40,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "width": "100%",
      },
      "largeTextBold": {
        "color": "black",
        "fontSize": 40,
        "fontWeight": "800",
      },
      "list": {
        "backgroundColor": "white",
        "borderColor": "lightgrey",
        "borderRadius": 4,
        "borderWidth": 1,
        "width": "80%",
      },
      "paddingBottom": 8,
      "paddingTop": 8,
      "smallButton": {
        "backgroundColor": "purple",
        "borderRadius": 5,
        "marginTop": 10,
        "padding": 10,
        "paddingHorizontal": 30,
      },
      "text": {
        "color": "black",
      },
      "textBold": {
        "color": "black",
        "fontWeight": "700",
      },
      "title": {
        "alignSelf": "center",
        "color": "black",
        "fontSize": 50,
        "fontWeight": "900",
        "paddingVertical": 10,
      },
    }
  }
>
  <View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          
You have no stock accounts

        </Text>
      }
      collapsable={false}
      data={null}
      getItem={[Function]}
      getItemCount={[Function]}
      handlerTag={3}
      handlerType="NativeViewGestureHandler"
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      style={
        {
          "paddingHorizontal": 8,
        }
      }
      viewabilityConfigCallbackPairs={[]}
      waitFor={
        [
          {
            "current": null,
          },
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <Text
          onLayout={[Function]}
          style={
            {
              "color": "black",
            }
          }
        >
          
You have no stock accounts

        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`<SuccessComponent /> listStocksScreen snapshot test 1`] = `
<View
  style={
    {
      "backArrow": {
        "color": "purple",
        "fontSize": 30,
        "fontWeight": "900",
        "paddingVertical": 10,
      },
      "btn": {
        "alignItems": "center",
        "borderRadius": 10,
        "margin": 10,
        "padding": 10,
        "width": "40%",
      },
      "button": {
        "backgroundColor": "purple",
        "borderRadius": 25,
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": "4%",
        "paddingHorizontal": "12%",
        "paddingVertical": "2%",
        "textAlign": "center",
        "width": "75%",
      },
      "buttonWide": {
        "backgroundColor": "purple",
        "borderRadius": 10,
        "marginVertical": 10,
        "padding": 20,
      },
      "container": {
        "backgroundColor": "white",
        "flex": 1,
      },
      "input": {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "height": 40,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 10,
        "width": "100%",
      },
      "largeTextBold": {
        "color": "black",
        "fontSize": 40,
        "fontWeight": "800",
      },
      "list": {
        "backgroundColor": "white",
        "borderColor": "lightgrey",
        "borderRadius": 4,
        "borderWidth": 1,
        "width": "80%",
      },
      "paddingBottom": 8,
      "paddingTop": 8,
      "smallButton": {
        "backgroundColor": "purple",
        "borderRadius": 5,
        "marginTop": 10,
        "padding": 10,
        "paddingHorizontal": 30,
      },
      "text": {
        "color": "black",
      },
      "textBold": {
        "color": "black",
        "fontWeight": "700",
      },
      "title": {
        "alignSelf": "center",
        "color": "black",
        "fontSize": 50,
        "fontWeight": "900",
        "paddingVertical": 10,
      },
    }
  }
>
  <View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          
You have no stock accounts

        </Text>
      }
      collapsable={false}
      data={
        [
          {
            "access_token": "access-sandbox-9cf73c44-d4f5-4a9c-84ce-6203d35aa40d",
            "account_id": "ALxMKPjkkdtZVpLBdXx8ia9prXRBmwh67wXWd",
            "balance": "266.23",
            "balance_currency": "GBP",
            "institution_id": "ins_115616",
            "institution_logo": "iVBORw0KGgoAAAANSUhEUgAAAJgAAACYCAMAAAAvHNATAAAAOVBMVEVHcEyZHCCYGx////+ZGx+ZGx+ZHCCcICKZGx+cKCyTFRjdtLWpSEv58PDnysu4aGrx3t7SnZ7FhIaoTrn+AAAACnRSTlMAR///tZt1INj4D9SNwAAABVVJREFUeNrNnNl2ozAMhsGDsS22wPs/7JCkbYIDshZD7Ks5Z6D90aet3qpKO1zd2qZpvA/GBO/Xf9q2dtU3h1sVeXMw/KrPfUfUoaY3dReLqymiXuLqq2zFUPWj7Xy7CVRdoc1ZIx/BuhJlPcYp0vSyTpGWR1Z+aW0w2UZo86Utb7IOXxdGMS/POpgTRlAbrTUnDZ2nucacNhpXGkY9ztacPIQ4rTl9WImuxlwwmkJ18ZU5by4a3pWpi6nsQl2rsuL8i+1nF+siK7Pm8mGLyPfCGlCbr4xk3XThO8KCK83xiQHQmq+NtkAHS7rZtxws6WZRBoPESP4q5ouWCnLp0DEllME87o8uMGFGpRum8R82xpTB+nnYfbFjlvOPiASzDIiwoU+ZDPrb3osLsCJzz/NhwpQtSS+DsGP0IQDL/+3uNy8alusP2LFZx6vm7oBGp2B5/wGfX4YGjaPWIugHDcuVzkAnuVeZ3HHQ61ia2OQd/jWO2B0CZjISy5lD8sPLHBLznYpl7GS3kHjHEdtpgEnHcmaRjEzmUC+5aVjGBp+SbzhqG4a5/5xmeWPE5Ef693iSVLCMY6dLf0lDI3lPRUgxT9bLhUvynWXLzt50liMvJrcsPQ8Hg6WA5Fv3k+70xSwlJF8No+UmSQbLjk/ylco8N7S2vys3yT+WjpC+OxnLuGrQSP7GJeGPXKwsoSxFJH/jkjLtFG4iltFrRJK/Tkaa1+wk9VJK8ulkjvKkjGVUZckkn05GmkcBEcvopZms6+FktJlNpMUYDgiBmOTTyWgzYliLcWQJOclHh0Gc00dajMPfOEpJPryf+CjWLu4zitM+g6QxxKBMlKV9W0QVlkPyHpbUSUQwbJZykvcGgzzrirUYe5RUJNd8QV4H4bJUkVzzBXn+HJCytPNb48d5JNd8QZ/Yx8rSJycdSZYwrMX4tEf0GUySqzDOmimHJahIrhmWIYzFMjLvxBbGWHMABsvoI0YmSRMqztNIWYpZRo9ySa41iSWsp7KMjdvzhbGWj5AWI+rne/rfeAcoWTsZkLK0Zakm6ZnCkLK0ZTkqSXruki6NZfQBfJK8zE9nGT3GJ8kXRmQ5cmfdP4Xxtn9gLcaLZUzS8IflLs8jZenFUk9ybRS56/NIWfpjuS3gEpJra+24r6RZRmlfQvI+R2CysRzC7iMSkoa/iQ3SLDOQ9IJNM0iL8cNyK32UkGwE28WQFuNpnIjkAqJsIdjOg7QYD5Zbk4pIPqah2GGJlKXnOrye5HN22LNZDhjLCLWIpBfuSURZbg0qI2lFe8YgwXLUk2yrSuRkKMvtf4pI/i4MsvcJIwslSw6SXrrxFSlL3baWykha8QbA47I0TBlI1uI938csh1s+koLkj81i5IpJWVwiLUa2mJSxnCm6tCQlLPsrSApYouv3OpLb7YD8VLacRdJWlcpkBPdXdDwak83nkLTUrYCSyQINSac+KZL0MhFJW1VakyW9bMliMMHZh0RdEpG0OTbzg0H9fwRtDpOnfxTmBLqkrznxpt0ji1VJ3YkRZPd6Z/jC6oyHyw6VCWLS5jzMcmQzAUnszJTg+M+BMkF2rTMfmIJ+zEKyzX7EbO9oyY1NsjnjUB70800Zk8lDeaJTZgAQpm7QZNf6tIOfYN7Mxo7J9syjsm9m67JlsDxnLFek/TLyY/KKY88AZuqYMUk/9q87WA+GF9mMA+zlHvkv9pKEcq+VKPgijnKvLin3spdyr8cp90Khcq9gKvjSqnKv+Sr3YrSCr5Ir+PK9bNKaMm9SLPTuSXvqbZ1lXiJa8LWrJV9UW/LVvi9x5V2G/CbvnOuj/wNfYsKPslREjwAAAABJRU5ErkJggg==",
            "institution_name": "Vanguard",
            "name": "Plaid IRA",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      style={
        {
          "paddingHorizontal": 8,
        }
      }
      viewabilityConfigCallbackPairs={[]}
      waitFor={
        [
          {
            "current": null,
          },
          {
            "current": null,
          },
        ]
      }
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "opacity": 1,
                "padding": 20,
                "text": "black",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJgAAACYCAMAAAAvHNATAAAAOVBMVEVHcEyZHCCYGx////+ZGx+ZGx+ZHCCcICKZGx+cKCyTFRjdtLWpSEv58PDnysu4aGrx3t7SnZ7FhIaoTrn+AAAACnRSTlMAR///tZt1INj4D9SNwAAABVVJREFUeNrNnNl2ozAMhsGDsS22wPs/7JCkbYIDshZD7Ks5Z6D90aet3qpKO1zd2qZpvA/GBO/Xf9q2dtU3h1sVeXMw/KrPfUfUoaY3dReLqymiXuLqq2zFUPWj7Xy7CVRdoc1ZIx/BuhJlPcYp0vSyTpGWR1Z+aW0w2UZo86Utb7IOXxdGMS/POpgTRlAbrTUnDZ2nucacNhpXGkY9ztacPIQ4rTl9WImuxlwwmkJ18ZU5by4a3pWpi6nsQl2rsuL8i+1nF+siK7Pm8mGLyPfCGlCbr4xk3XThO8KCK83xiQHQmq+NtkAHS7rZtxws6WZRBoPESP4q5ouWCnLp0DEllME87o8uMGFGpRum8R82xpTB+nnYfbFjlvOPiASzDIiwoU+ZDPrb3osLsCJzz/NhwpQtSS+DsGP0IQDL/+3uNy8alusP2LFZx6vm7oBGp2B5/wGfX4YGjaPWIugHDcuVzkAnuVeZ3HHQ61ia2OQd/jWO2B0CZjISy5lD8sPLHBLznYpl7GS3kHjHEdtpgEnHcmaRjEzmUC+5aVjGBp+SbzhqG4a5/5xmeWPE5Ef693iSVLCMY6dLf0lDI3lPRUgxT9bLhUvynWXLzt50liMvJrcsPQ8Hg6WA5Fv3k+70xSwlJF8No+UmSQbLjk/ylco8N7S2vys3yT+WjpC+OxnLuGrQSP7GJeGPXKwsoSxFJH/jkjLtFG4iltFrRJK/Tkaa1+wk9VJK8ulkjvKkjGVUZckkn05GmkcBEcvopZms6+FktJlNpMUYDgiBmOTTyWgzYliLcWQJOclHh0Gc00dajMPfOEpJPryf+CjWLu4zitM+g6QxxKBMlKV9W0QVlkPyHpbUSUQwbJZykvcGgzzrirUYe5RUJNd8QV4H4bJUkVzzBXn+HJCytPNb48d5JNd8QZ/Yx8rSJycdSZYwrMX4tEf0GUySqzDOmimHJahIrhmWIYzFMjLvxBbGWHMABsvoI0YmSRMqztNIWYpZRo9ySa41iSWsp7KMjdvzhbGWj5AWI+rne/rfeAcoWTsZkLK0Zakm6ZnCkLK0ZTkqSXruki6NZfQBfJK8zE9nGT3GJ8kXRmQ5cmfdP4Xxtn9gLcaLZUzS8IflLs8jZenFUk9ybRS56/NIWfpjuS3gEpJra+24r6RZRmlfQvI+R2CysRzC7iMSkoa/iQ3SLDOQ9IJNM0iL8cNyK32UkGwE28WQFuNpnIjkAqJsIdjOg7QYD5Zbk4pIPqah2GGJlKXnOrye5HN22LNZDhjLCLWIpBfuSURZbg0qI2lFe8YgwXLUk2yrSuRkKMvtf4pI/i4MsvcJIwslSw6SXrrxFSlL3baWykha8QbA47I0TBlI1uI938csh1s+koLkj81i5IpJWVwiLUa2mJSxnCm6tCQlLPsrSApYouv3OpLb7YD8VLacRdJWlcpkBPdXdDwak83nkLTUrYCSyQINSac+KZL0MhFJW1VakyW9bMliMMHZh0RdEpG0OTbzg0H9fwRtDpOnfxTmBLqkrznxpt0ji1VJ3YkRZPd6Z/jC6oyHyw6VCWLS5jzMcmQzAUnszJTg+M+BMkF2rTMfmIJ+zEKyzX7EbO9oyY1NsjnjUB70800Zk8lDeaJTZgAQpm7QZNf6tIOfYN7Mxo7J9syjsm9m67JlsDxnLFek/TLyY/KKY88AZuqYMUk/9q87WA+GF9mMA+zlHvkv9pKEcq+VKPgijnKvLin3spdyr8cp90Khcq9gKvjSqnKv+Sr3YrSCr5Ir+PK9bNKaMm9SLPTuSXvqbZ1lXiJa8LWrJV9UW/LVvi9x5V2G/CbvnOuj/wNfYsKPslREjwAAAABJRU5ErkJggg==",
                  }
                }
                style={
                  {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "black",
                      "fontFamily": "sans-serif",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                    
                  Plaid IRA
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "black",
                        "fontFamily": "sans-serif",
                        "fontSize": 15,
                      },
                      {
                        "fontSize": 11,
                      },
                    ]
                  }
                >
                     
                  Vanguard
                   • 
                  GBP
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "black",
                        "fontFamily": "sans-serif",
                        "fontSize": 15,
                      },
                      {
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "height": 19,
                      },
                    ]
                  }
                >
                  £
                  266
                </Text>
                <View
                  style={
                    {
                      "height": 16,
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "black",
                          "fontFamily": "sans-serif",
                          "fontSize": 15,
                        },
                        {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    .
                    23
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
